packages/legend-application-data-cube/src/stores/LegendDataCubeDataCubeEngine.ts

```diff
@@ -342,6 +367,12 @@ export class LegendDataCubeDataCubeEngine extends DataCubeEngine {
       result = await this._runQuery(query, source.model);
     } else if (source instanceof LegendQueryDataCubeSource) {
       query.parameters = source.lambda.parameters;
+      // since let function was added to the end of the lambda body arrary before
+      // we need to get it by index 1 here
+      const letFunction = source.lambda.body.at(1);
```

Comment:
related to the above comment, isnt this suppose to be at the beginning?

Reply:
as coded below, 
```
      if (letFunction) {
        query.body = [letFunction, ...query.body];
      }
```

it's added to the beginning.

However, i can switch all to the beginning

Reply:
Does this assume there will only be one let statement ? 
what if there is 2. 
let x = now();
let y = now();

would this work and capture both values ?

Reply:
mhm okay makes sense, the beginning part

Reply:
does it make sense to maybe adding another proeprty on LegendQuerySource -> `letParameterValues` and populate that at the beginining ?